public/src/main/java/com/revenuecat/purchases/models/PricingPhase.kt

```diff
@@ -28,4 +28,27 @@ data class PricingPhase(
      * [Price] of the [PricingPhase]
      */
     val price: Price
-) : Parcelable
+) : Parcelable {
+
+    /**
+     * Indicates how the pricing phase is charged for FINITE_RECURRING pricing phases
+     */
+    val paymentMode: PaymentMode?
+        get() {
+            // billingCycleCount is null for INFINITE_RECURRING or NON_RECURRING recurrence modes
+            // but validating for FINITE_RECURRING anyway
+            if (recurrenceMode != RecurrenceMode.FINITE_RECURRING) {
+                return null
+            }
+
+            return if (price.amountMicros == 0L) {
+                PaymentMode.FREE_TRIAL
+            } else if (billingCycleCount == 1) {
+                PaymentMode.PAY_UP_FRONT
```

Comment:
This feels a bit weird... I guess it's technically paying up front, but only a single billing cycle. Users are not paying up front for multiple periods as it's supposed to mean in iOS... 

I can understand it if this cycle is longer than the others though, (like say a 1 year cycle paid up front and then going to a monthly subscription) so I guess it's ok.

Reply:
It does feel weird but that is kind of how it works on Google Play Console 👇 

"Single Payment" is pay upfront (which is 1 billing cycle) and "Discounted recurring payment" allows you to enter a number of months but it has to be 2 or more 🤷‍♂️ 

<img width="646" alt="Screenshot 2023-04-04 at 8 51 35 PM" src="https://user-images.githubusercontent.com/401294/230097316-16999fa4-5bf7-4565-941c-4073d2aa1ea4.png">

Reply:
We could also consider naming the constants the same as they are on GPC (`FREE_TRIAL`, `SINGLE_PAYMENT`, `DISCOUNTED_RECURRING_PAYMENT`).
For the hybrids though the current naming which mirrors Apple's is probably better

Reply:
As for naming it using google's naming... I kinda like Apple's naming more, so I'd vote to keep it as is